cmd/thanos/main.go

```diff
@@ -143,8 +142,26 @@ func main() {
 	{
 		ctx := context.Background()
 
-		var closeFn func() error
-		tracer, closeFn = tracing.NewOptionalGCloudTracer(ctx, logger, *gcloudTraceProject, *gcloudTraceSampleFactor, *debugName)
+		var closer io.Closer
+		var confContentYaml []byte
+		confContentYaml, err = tracingConfig.Content()
+
+		if len(confContentYaml) == 0 {
+			level.Info(logger).Log("msg", "Tracing will be disabled")
+			tracer, closer, err = provider.NoopTracer()
```

Comment:
Can we adjust Noop Tracer to just return without `err` and `closer`? It's not needed (:

Reply:
then we can move below error to be only for `provider.NewTracer`

Reply:
or `tracing.NewTracer` as `provider` is not very explicit as commented above